paimon-core/src/main/java/org/apache/paimon/table/AbstractFileStoreTable.java

```diff
@@ -289,6 +295,11 @@ public ExpireSnapshots newExpireSnapshots() {
 
     @Override
     public TableCommitImpl newCommit(String commitUser) {
+        // Compatibility with previous design, the main branch is written by default
+        return newCommit(commitUser, DEFAULT_MAIN_BRANCH);
+    }
+
+    public TableCommitImpl newCommit(String commitUser, String branchName) {
```

Comment:
This is not implemented completely right? Because there isn't any other caller for this method and pass a `branchName` besides `DEFAULT_MAIN_BRANCH`. Are you planning to implement this in the next pr?

Reply:
In AbstractFileStoreTable line 320, this method is invoked.

Reply:
Example : In FileStoreTableTestBase line 1284

Reply:
I mean if we've created a branch for Paimon table through procedure, how to really write to this custom branch in a Flink job? It seems like the committer operator in Flink job is only able to write to the default branch currently.

Reply:
Maybe there should be a option for Flink connector in `FlinkConnectorOptions` to set a writing branch for Flink jobs, and in Flink sink this option could be recognized to create a branched commit. But I think these could be implemented in the next PR for the scope for this PR is big enough, breaking code to PRs can make the code more clear to track and review.

Reply:
I agree with your suggestion, and I plan to support flink and spark read and write after this pr merge. As you suggest, this is more conducive to review.